OpenSim/Common/Property.h

```diff
@@ -938,6 +952,88 @@ readSimplePropertyFromStream(std::istream& in)
        return SimTK::readUnformatted(in, values);
 }
 
+// We provide specializations for all types that contain floats to manage
+// the number of digits displayed. We rely on the default setting of how
+// `stringstream`s handle floats, where `precision` is the number of digits
+// printed, and the `stream` automatically determines whether float or
+// exponential notation is used.
+template <> inline void SimpleProperty<double>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << values[i];
+    }
+}
+
+template <> inline void SimpleProperty<SimTK::Vec3>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << "(" << values[i][0] << " ";
+        o << values[i][1] << " ";
+        o << values[i][2] << ")";
+    }
+}
+
+template <> inline void SimpleProperty<SimTK::Vec6>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << "(" << values[i][0] << " ";
+        o << values[i][1] << " ";
+        o << values[i][2] << " ";
+        o << values[i][3] << " ";
+        o << values[i][4] << " ";
+        o << values[i][5] << ")";
+    }
+}
```

Comment:
(from discussion with @aseth1) You might be able to use one implementation for both `Vec3` and `Vec6` (actually, for `Vec<N>`). See the function template specialization of `writeUnformatted()` for `Vec<N>`.

@aseth1 said there is at least one `Vec2` property; so right now, `Vec2` would be displayed in full precision.

Reply:
After discussion with @aseth1 and @chrisdembia, the plan above isn't allowed in C++. Will make these functions a free function in the OpenSim namespace, which then will be in parallel with Simbody's `writeUnformatted()` function.